homeassistant/components/cover/mqtt.py

```diff
@@ -145,6 +160,10 @@ def __init__(self, name, state_topic, command_topic, tilt_command_topic,
         self._tilt_max = tilt_max
         self._tilt_optimistic = tilt_optimistic
         self._tilt_invert = tilt_invert
+        self._position_topic = position_topic
+        self._set_position_template = set_position_template
+        if set_position_template is not None:
+            self._set_position_template.hass = hass
```

Comment:
Can you put this logic in `async_setup_platform` so we don't need to pass `hass` in? Let's make it look the same as `value_template`.